### PR TITLE
feat: add get_course_jobs util method

### PR DESCRIPTION
**Pull request by muhammad-ammar**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [ ] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

File: taxonomy/utils.py [hunk 1]
```diff
@@ -207,3 +207,23 @@ def get_blacklisted_course_skills(course_key, prefetch_skills=True):
...
+    Arguments:
+        course_key (str): Key of the course whose course skills need to be returned.
+
+    Returns:
+        list: A list of all the course jobs.
+    """
+    course_jobs = set()
+    course_skills = get_whitelisted_course_skills(course_key)
+    for course_skill in course_skills:
+        job_skills = JobSkills.objects.select_related('skill', 'job').filter(skill=course_skill.skill)
```
Comment (saleem-latif) [inline taxonomy/utils.py:225]: would it better to improve get all `JobSkills` in a single query? 
```
job_skills = JobSkills.objects.select_related('skill', 'job').filter(skill__in=[course_skill.skill for course_skill in course_skills])
```
and remove the outer for loop.

File: taxonomy/utils.py [hunk 2]
```diff
@@ -207,3 +207,23 @@ def get_blacklisted_course_skills(course_key, prefetch_skills=True):
...
+        for job_skill in job_skills:
```
Comment (saleem-latif) [inline taxonomy/utils.py:226]: We can also remove this loop using list comprehension.
```
return [job_skill.job.name for job_skill in job_skills]
```

**Review by saleem-latif — APPROVED** [review]
Just a couple of comments/suggestion, once those are addressed 👍🏼

**Comment by muhammad-ammar** [comment]
@saleem-latif Addressed your review. Please have a look.

File: taxonomy/utils.py [hunk 3]
```diff
@@ -207,3 +207,23 @@ def get_blacklisted_course_skills(course_key, prefetch_skills=True):
...
+        list: A list of all the course jobs.
+    """
+    course_skills = get_whitelisted_course_skills(course_key)
+    job_skills = JobSkills.objects.select_related(
+        'skill',
+        'job',
+    ).filter(
+        skill__in=[course_skill.skill for course_skill in course_skills]
+    )
+    return list({job_skill.job.name for job_skill in job_skills})
```
Comment (saleem-latif) [inline taxonomy/utils.py:229]: why create a set and the convert it to list?  you can either return the set or create a list in the first place.
```
return {job_skill.job.name for job_skill in job_skills}
# OR
return [job_skill.job.name for job_skill in job_skills]
```

**Review by saleem-latif — APPROVED** [review]

**Pull request merged**